src/js/common/utils/lodash-replacements/set.js

```diff
@@ -0,0 +1,70 @@
+import deconstructPath from './deconstructPath';
+import cloneDeep from './cloneDeep';
+
+/**
+ * Same as `set`, but uses the level param to determine when to clone and give a more helpful error message.
+ *
+ * @param {Array|Obect} object
+ * @param {Array|string} path
+ * @param {*} value
+ * @param {Number} level  How many times we've recursed
+ * @return {Object} A new object with the appropriate value set
+ */
+function baseSet(path, value, object, level = 0) {
+  let arrayPath = Array.isArray(path) ? path : deconstructPath(path);
+
+  if (!arrayPath.length) {
+    // We're at the end of our path; time to assign
+    return value;
+  }
+
+  // Do some one-time prep work
+  let newObj = object;
+  if (level === 0) {
+    // Only clone the whole object once
+    newObj = cloneDeep(object);
+    // The following path segments are not allowed
+    arrayPath = arrayPath.filter(e => e !== undefined && e !== null);
```

Comment:
Could we throw an error when this occurs instead? Bad path segments seems like a case of developer error.

Reply:
Yeah, I should also pull that bit out into a helper function to be used in `get()` as well, methinks.